### PR TITLE
volume attach by FC protocol

### PR DESCRIPTION
**Pull request by lasshpanzer**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/dock/proto/dock.proto [hunk 1]
```diff
@@ -211,6 +211,8 @@ message CreateAttachmentOpts {
     string driverName = 7;
     // The Context
     string context = 8;
+	// The protocol
```
Comment (leonwanghui) [inline pkg/dock/proto/dock.proto]: Please change the format.

File: pkg/model/volume.go [hunk 2]
```diff
@@ -107,6 +107,9 @@ type VolumeAttachmentSpec struct {
 
 	// See details in `ConnectionInfo`
 	ConnectionInfo `json:"connectionInfo,omitempty"`
+
+	// The protocl
+	Protocol string `json:"protocol,omitempty"`
```
Comment (leonwanghui) [inline pkg/model/volume.go]: To me, it's unacceptable for users to specify access protocol when creating volume attachment.

Comment (lasshpanzer) [reply pkg/model/volume.go]: Protocol here is provided by pool.

File: pkg/dock/proto/dock.proto [hunk 3]
```diff
@@ -228,6 +230,8 @@ message DeleteAttachmentOpts {
     string driverName = 5;
     // The Context
     string context = 6;
+	// The protocol
```
Comment (leonwanghui) [inline pkg/dock/proto/dock.proto]: Please change the format

File: osdsctl/cli/volumeattachment.go [hunk 4]
```diff
@@ -92,7 +92,7 @@ func volumeAttachmentCreateAction(cmd *cobra.Command, args []string) {
 		Fatalln(HttpErrStrip(err))
 	}
 	keys := KeyList{"Id", "CreatedAt", "UpdatedAt", "TenantId", "UserId", "HostInfo", "ConnectionInfo",
-		"Mountpoint", "Status", "VolumeId"}
+		"Mountpoint", "Status", "VolumeId", "Protocol"}
```
Comment (leonwanghui) [inline osdsctl/cli/volumeattachment.go]: I suggest you change `Protocol` to `AccessProtocol`

File: pkg/controller/controller.go [hunk 5]
```diff
@@ -335,6 +349,7 @@ func (c *Controller) CreateVolumeAttachment(ctx *c.Context, in *model.VolumeAtta
 			Host:      in.Host,
 			Initiator: in.Initiator,
 		},
+		Protocol:   protocol,
```
Comment (leonwanghui) [inline pkg/controller/controller.go]: Change `Protocol` to `AccessProtocol`

File: pkg/model/volume.go [hunk 6]
```diff
@@ -108,6 +108,9 @@ type VolumeAttachmentSpec struct {
 
 	// See details in `ConnectionInfo`
 	ConnectionInfo `json:"connectionInfo,omitempty"`
+
+    // The protocl
```
Comment (leonwanghui) [inline pkg/model/volume.go]: Please update the format.

**Pull request merged**